src/topology/continuous_function/zero_at_infty.lean

```diff
@@ -0,0 +1,318 @@
+
+instance : has_neg (α →C₀ β) :=
+⟨λ f, ⟨-f, by simpa only [neg_zero] using (zero_at_infty f : tendsto f (cocompact α) (𝓝 0)).neg⟩⟩
+
+@[simp] lemma coe_neg : ⇑(-f) = -f := rfl
+lemma neg_apply : (-f) x = -f x := rfl
+
+instance : has_sub (α →C₀ β) :=
+⟨λ f g, ⟨f - g,
+begin
+  rw sub_eq_add_neg,
+  simpa only [add_zero] using ((zero_at_infty f).add (zero_at_infty (-g)) :
+    tendsto (λ x, f x + (-g) x) (cocompact α) (𝓝 (0 + 0))),
+end⟩⟩
+
+@[simp] lemma coe_sub : ⇑(f - g) = f - g := rfl
+lemma sub_apply : (f - g) x = f x - g x := rfl
+
+@[simp] lemma coe_zsmul_rec : ∀ z, ⇑(zsmul_rec z f) = z • f
+| (int.of_nat n) := by rw [zsmul_rec, int.of_nat_eq_coe, coe_nsmul_rec, coe_nat_zsmul]
+| -[1+ n] := by rw [zsmul_rec, zsmul_neg_succ_of_nat, coe_neg, coe_nsmul_rec]
+
+instance has_int_scalar : has_scalar ℤ (α →C₀ β) :=
+⟨λ n f, ⟨n • f, by simpa using (zsmul_rec n f).zero_at_infty'⟩⟩
+
+instance : add_group (α →C₀ β) :=
+fun_like.coe_injective.add_group _ coe_zero coe_add coe_neg coe_sub (λ _ _, rfl) (λ _ _, rfl)
+
+end add_group
+
+instance [add_comm_group β] [topological_add_group β] : add_comm_group (α →C₀ β) :=
+fun_like.coe_injective.add_comm_group _ coe_zero coe_add coe_neg coe_sub (λ _ _, rfl) (λ _ _, rfl)
+
+instance [has_zero β] {R : Type*} [has_zero R] [smul_with_zero R β]
+  [has_continuous_const_smul R β] : has_scalar R (α →C₀ β) :=
+⟨λ r f, ⟨r • f, by simpa [smul_zero] using
+  (zero_at_infty f : tendsto f (cocompact α) (𝓝 0)).const_smul r⟩⟩
+
+@[simp] lemma coe_smul [has_zero β] {R : Type*} [has_zero R] [smul_with_zero R β]
+  [has_continuous_const_smul R β] (r : R) (f : α →C₀ β) : ⇑(r • f) = r • f := rfl
+
+lemma smul_apply [has_zero β] {R : Type*} [has_zero R] [smul_with_zero R β]
+  [has_continuous_const_smul R β] (r : R) (f : α →C₀ β) (x : α) : (r • f) x = r • f x := rfl
+
+instance [has_zero β] {R : Type*} [has_zero R] [smul_with_zero R β]
+  [has_continuous_const_smul R β] : smul_with_zero R (α →C₀ β) :=
+function.injective.smul_with_zero ⟨_, coe_zero⟩ fun_like.coe_injective coe_smul
+
+instance [has_zero β] {R : Type*} [monoid_with_zero R] [mul_action_with_zero R β]
+  [has_continuous_const_smul R β] : mul_action_with_zero R (α →C₀ β) :=
+function.injective.mul_action_with_zero ⟨_, coe_zero⟩ fun_like.coe_injective coe_smul
+
+instance [add_comm_monoid β] [has_continuous_add β] {R : Type*} [comm_semiring R] [module R β]
+  [has_continuous_const_smul R β] : module R (α →C₀ β) :=
+function.injective.module R ⟨_, coe_zero, coe_add⟩ fun_like.coe_injective coe_smul
+
+instance [non_unital_semiring β] [has_continuous_add β] [has_continuous_mul β] :
+  non_unital_semiring (α →C₀ β) :=
+fun_like.coe_injective.non_unital_semiring _ coe_zero coe_add coe_mul (λ _ _, rfl)
+
+end algebraic_structure
+
+section metric
+
+open metric set
+
+variables [metric_space β] [has_zero β] [zero_at_infty_continuous_map_class F α β]
+
+protected lemma bounded (f : F) : ∃ C, ∀ x y : α, dist ((f : α → β) x) (f y) ≤ C :=
+begin
```

Comment:
There's a probably common idea between this lemma and `continuous.exists_forall_le`, are you able to put your finger on it and extract it?